tests/lib.rs

```diff
@@ -465,6 +466,55 @@ async fn it_works_with_as_pin_mut() {
     assert_eq!(Future1::new().await, 10);
 }
 
+#[test]
+fn it_works_with_as_pin_drop() {
+    struct Trap {
+        data: String,
+        ptr: *const String,
+        _marker: std::marker::PhantomPinned,
+    }
+    impl Drop for Trap {
+        fn drop(&mut self) {
+            let this = unsafe { Pin::new_unchecked(self) };
```

Comment:
Technically it's okay to not do this explicitly. It serves [as a reminder that self was pinned](https://doc.rust-lang.org/std/pin/index.html#drop-implementation). Even if self was not pinned before, it doesn't hurt to assert that it will not move for the rest of the drop impl, since pinning guarantees end after drop has been called anyway.

Reply:
Thanks. Now I understood the intention.